lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// StoreOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateStoreToSDFG(StoreOp &op, JsonEmitter &jemit) {
+  jemit.startObject();
+  jemit.printKVPair("type", "MultiConnectorEdge");
+
+  jemit.startNamedObject("attributes");
+  jemit.startNamedObject("data");
+  jemit.printKVPair("type", "Memlet");
+  jemit.startNamedObject("attributes");
+
+  jemit.printKVPair("volume", 1);
+  jemit.printKVPair("num_accesses", 1);
+
+  if (GetAccessOp aNode = dyn_cast<GetAccessOp>(op.arr().getDefiningOp())) {
+    jemit.printKVPair("data", aNode.getName());
+  } else {
+    mlir::emitError(op.getLoc(), "Array must be defined by a GetAccessOp");
+    return failure();
+  }
+
+  jemit.startNamedObject("subset");
+  jemit.printKVPair("type", "Range");
+  jemit.startNamedList("ranges");
+
+  if (ArrayAttr syms = op->getAttr("indices").cast<ArrayAttr>()) {
+    for (Attribute sym : syms.getValue()) {
+      if (StringAttr sym_str = sym.cast<StringAttr>()) {
+        jemit.startObject();
+        jemit.printKVPair("start", sym_str.getValue());
+        jemit.printKVPair("end", sym_str.getValue());
+        jemit.printKVPair("step", 1);
+        jemit.printKVPair("tile", 1);
+        jemit.endObject();
+      } else {
+        mlir::emitError(op.getLoc(), "'indices' must consist of StringAttr");
+        return failure();
+      }
+    }
+  } else {
+    mlir::emitError(op.getLoc(), "'indices' must be an ArrayAttr");
+    return failure();
+  }
+
+  jemit.endList();   // ranges
+  jemit.endObject(); // subset
+
+  jemit.startNamedObject("dst_subset");
+  jemit.printKVPair("type", "Range");
+  jemit.startNamedList("ranges");
+
+  if (ArrayAttr syms = op->getAttr("indices").cast<ArrayAttr>()) {
+    for (Attribute sym : syms.getValue()) {
+      if (StringAttr sym_str = sym.cast<StringAttr>()) {
+        jemit.startObject();
+        jemit.printKVPair("start", sym_str.getValue());
+        jemit.printKVPair("end", sym_str.getValue());
+        jemit.printKVPair("step", 1);
+        jemit.printKVPair("tile", 1);
+        jemit.endObject();
+      } else {
+        mlir::emitError(op.getLoc(), "'indices' must consist of StringAttr");
+        return failure();
+      }
+    }
+  } else {
+    mlir::emitError(op.getLoc(), "'indices' must be an ArrayAttr");
+    return failure();
+  }
+
+  jemit.endList();   // ranges
+  jemit.endObject(); // dst_subset
+
+  jemit.endObject(); // attributes
+  jemit.endObject(); // data
+  jemit.endObject(); // attributes
+
+  if (sdir::CallOp call = dyn_cast<sdir::CallOp>(op.val().getDefiningOp())) {
+    TaskletNode aNode = call.getTasklet();
+    jemit.printKVPair("src", aNode.ID());
+    jemit.printKVPair("src_connector", "__out");
+  } else if (arith::ConstantOp con =
+                 dyn_cast<arith::ConstantOp>(op.val().getDefiningOp())) {
+    // TODO: Add constants
```

Comment:
necessary?

Reply:
Yes if we want to support snippets like this:

```
%1 = arith.constant 1 : i32
%c = sdir.call @add(%1, %1) : (i32, i32) -> i32
```

Reply:
but you collect all the constants beforehand, no? so this shouldn't matter

Reply:
Don't I still need to tell which constant goes to which connector in the tasklet?
Or for stores, where in the array to store it? I don't see how emitting the constant in `constant_prop` is enough

Reply:
constants don't need connectors in the graph, you can refer to them directly in tasklets or interstate edges.

Reply:
Does this also work for MLIR tasklets?
I would still need to emit a tasklet that "lacks" the connector where the constant would go in. So need to handle this case differently.
Same thing when storing a constant in an array

Reply:
You mean a constant that was defined outside of an MLIR tasklet, but used within it?

For the second case, you'd need to make a tasklet for that store (which can use the constant's name as is)

Reply:
Yes!
Can DaCe lift tasklets which generate a constant (such as `__out = dace.int64(5)`) to a SDFG constant?
If so I think the most elegant solution would be to disallow other dialects in a `StateOp` and therefore require explicit tasklets for constants. This way we can avoid handling special cases and getting involved with the arithmetic dialect for the translator. Besides, as you just said, constant stores need a tasklet anyway.

Reply:
1. Yes, there is a process called symbol promotion, which promotes an assignment tasklet to a symbolic value (not constant, though that should definitely be considered! great idea :) )
2. The elegant solution you are thinking of is actually the best one. It will also ensure that round-trips of SDFG->SDIR->SDFG would always stay the same. Let's go with that.